### DISCUSSION TITLE
RPC Provider/ Chain Addition

### DISCUSSION THREAD

**Discussion by GrizzlyWebDev** [wevm/wagmi#1051] [Q&A] [answered]
I was wondering if there is a way to support Binance Smart Chain since it is also EVM based. I saw the example with avax in the docs but it wasn't quite working with smart chain.

**Comment by GrizzlyWebDev** [comment, marked as answer]
I was being silly, following the avax example I was able to create a new chain file and import it into the configureChains function and everything is working properly. 

Only thing to note is smart chain does not support multicalls. So you can only use useContractRead and not useContractReads. Hope this helps someone else! 

```
in main.jsx or index.js

import { WagmiConfig, createClient, configureChains } from "wagmi";
import { MetaMaskConnector } from "wagmi/connectors/metaMask";
import { WalletConnectConnector } from "wagmi/connectors/walletConnect";
import { jsonRpcProvider } from "wagmi/providers/jsonRpc";
import { publicProvider } from "wagmi/providers/public";

const { chains, provider } = configureChains(
  [binanceChain],
  [publicProvider()],
  [
    jsonRpcProvider({
      rpc: (chain) => {
        if (chain.id !== binanceChain.id) return null;
        return { http: chain.rpcUrls.default };
      },
    }),
  ]
);

const client = createClient({
  connectors: [
    new MetaMaskConnector({ chains }),
    new WalletConnectConnector({
      options: {
        chains,
        qrcode: true,
        rpc: {
          56: "https://bsc-dataseed4.binance.org",
        },
      },
    }),
  ],
  autoConnect: true,
  chains,
  provider,
});
```

and in a separte typescript file still within the src folder : 

```

import { Chain } from "wagmi";

export const binanceChain: Chain = {
  id: 56,
  name: "Binance Smart Chain Mainnet",
  network: "binance-smart chain mainnet",
  nativeCurrency: {
    name: "Binance Chain Native Token",
    symbol: "BNB",
    decimals: 18,
  },
  rpcUrls: {
    binance: "https://bsc-dataseed4.binance.org",
    defibit: "https://bsc-dataseed4.defibit.io",
    ninicoin: "https://bsc-dataseed4.ninicoin.io",
    public: "https://bsc-dataseed4.binance.org",
    default: "https://bsc-dataseed4.binance.org",
  },
  blockExplorers: {
    default: { name: "Bscscan", url: "https://bscscan.com" },
  },
  testnet: false,
};

```

**Reply by ghost** [reply]
Hey, very nice. Thank you for this. I was able to get this working for BSC & cronos by doing something very similar. My configureChains is slightly different tho:

```
const { chains, provider } = configureChains(
  //have to add mainnet because cro defi wallet req's it. Need to prompt to switch to CRO if !chainId25
  [cronos, chain.mainnet],
  [jsonRpcProvider({ rpc: chain => ({ http: chain.rpcUrls.default }) })]
);
```

I left out the publicProvider. Do you think that's something I should leave in and as the first option in the provider array? Think I read somewhere that it could be unreliable. 

Cheers man

**Reply by GrizzlyWebDev** [reply]
The only reason I left it in is because for BSC there isn't a reliable "Infura" Equivalent. I also have a list of RPCs in my config just in case of rate limiting.

**Reply by ghost** [reply]
ah gotcha, thanks man. appreciate it

**Reply by jeftarmascarenhas** [reply]
This example isn't work with Wagmi new version
You can use two forms Small and complete implementation;
Small: https://gist.github.com/jeftarmascarenhas/a16fcded79494099115e43738e6a9bb1
Complete: https://gist.github.com/jeftarmascarenhas/3214feb7c6871253cf9486fd8ccacc40
